subworkflows/local/utils_nfcore_crisprseq_pipeline/main.nf

```diff
@@ -77,42 +77,65 @@ workflow PIPELINE_INITIALISATION {
+
     //
     // Create channel from input file provided through params.input
     //
-    Channel
-        .fromSamplesheet("input")
-        .multiMap {
-            meta, fastq_1, fastq_2, reference, protospacer, template ->
-                if (fastq_2) {
-                    files = [ fastq_1, fastq_2 ]
-                } else {
-                    files = [ fastq_1 ]
-                }
-                reads_targeted: [ meta.id, meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], files ]
-                reads_screening:[ meta + [ single_end:fastq_2?false:true ], files ]
-                reference:      [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], reference]
-                protospacer:    [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], protospacer]
-                template:       [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], template]
-        }
-        .set { ch_input }
+    if(params.input) {
+        Channel
+            .fromSamplesheet("input")
+            .multiMap {
+                meta, fastq_1, fastq_2, reference, protospacer, template ->
+                    if (fastq_2) {
+                        files = [ fastq_1, fastq_2 ]
+                    } else {
+                        files = [ fastq_1 ]
+                    }
+                    reads_targeted: [ meta.id, meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], files ]
+                    reads_screening:[ meta + [ single_end:fastq_2?false:true ], files ]
+                    reference:      [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], reference]
+                    protospacer:    [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], protospacer]
+                    template:       [meta - meta.subMap('condition') + [ single_end:fastq_2?false:true, self_reference:reference?false:true, template:template?true:false ], template]
+            }
+            .set { ch_input }
+            fastqc_screening = ch_input.reads_screening
+            fastqc_screening = ch_input.reads_screening
+            reference = ch_input.reference
+            protospacer = ch_input.protospacer
+            template = ch_input.template
+    } else {
+        ch_input = Channel.empty()
+    }
 
     //
     // Validate input samplesheet
     //
-    ch_input.reads_targeted
-    .groupTuple()
-    .map {
-        validateInputSamplesheet(it)
+    if(params.input) {
```

Comment:
I think you don't need the if here, because `reads_targeted` was initialised at the beginning as an empty channel.

Reply:
then I would move it up inside the previous if. I find it confusing to set reads_targeted in two different places